src/AbstractClient.php

```diff
@@ -354,6 +360,7 @@ protected function prepareHeaders(array $headers): array
     {
         $baseHeaders = [
             self::API_KEY => $this->apiKey,
+            'User-Agent' => (string) new UserAgentHeader(),
```

Comment:
using a class for this seems overkill
I'd write the `sprintf` right here

Reply:
so you will duplicate the sprintf code in the unit tests ?